packages/edit-widgets/src/components/widget-areas/index.js

```diff
@@ -10,8 +11,18 @@ import { withSelect } from '@wordpress/data';
 import WidgetArea from '../widget-area';
 
 function WidgetAreas( { areas, blockEditorSettings } ) {
+	const [ selectedArea, setSelectedArea ] = useState( 0 );
```

Comment:
do we really need this here? It feels like this could be local state for `WidgetArea`

Reply:
The widget area controls which of the areas contains a selected block, when a new area is selected other becomes unselected, each WidgetArea is a sibling of each other and siblings can not "communicate" between each other. This local state in WidgetAreas that is the parent of each WidgetArea allows the WidgetArea components to communicate with each other, when one becomes selected the others become unselected and the corresponding effects get executed, I don't think is possible to make this state part of the WidgetArea component.

Reply:
ok makes sense.